File: go/pipeline.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// PipelineService — V3 defense pipeline execution.
type PipelineService struct {
	t *internal.Transport
}

func (s *PipelineService) Run(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/pipeline/run", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: python/src/aegis/resources/pipeline.py
"""Pipeline resource — V3 defense-pipeline execution endpoint.

Server contracts:

- ``POST /v3/pipeline/run`` — run a composed defense pipeline.
"""

from __future__ import annotations

from typing import Any, Dict

from aegis.resources._base import AsyncResource, SyncResource


class SyncPipeline(SyncResource):

    def run(self, body: Dict[str, Any]) -> Dict[str, Any]:
        return self._post("/v3/pipeline/run", json=body)


class AsyncPipeline(AsyncResource):

    async def run(self, body: Dict[str, Any]) -> Dict[str, Any]:
        return await self._post("/v3/pipeline/run", json=body)
